### PR TITLE
Feat/45 44 improve parsing config variable

### LINKED ISSUE #44: Add End-to-End (E2E) Tests
**Description:**  
The project currently lacks comprehensive end-to-end (E2E) tests for the runner. E2E tests are needed to ensure that the pipeline functions correctly as a whole, from data ingestion to final validation, covering all integrated components and their interactions.

**Expected Outcome:**  
A suite of E2E tests is added (e.g., in `tests/e2e/`) to verify the complete data pipeline flow, including data cleaning, processing, aggregation, and validation, for various input scenarios.

### LINKED ISSUE #45: Refactor and Improve Cleaning and Parsing of `config.yaml` and CLI Variables
**Description:**  
The current logic for loading, cleaning, and parsing the `config.yaml` file and CLI variables in the pipeline engine is fragmented and could be improved for clarity, maintainability, and robustness. Parsing is performed in multiple places, and error handling is inconsistent. There is also a need to better document and test the type conversion and precedence rules between config file values and CLI variables.

**Expected Outcome:**  
- Centralize and refactor the parsing logic for both `config.yaml` and CLI variables.
- Ensure consistent type conversion (e.g., string, int, bool, list) and error handling.
- Clearly document the precedence and merging rules.
- Add or improve tests for edge cases and invalid input.
- Update docstrings and comments for clarity.

### PR DESCRIPTION
**Pull request by BaptisteLacroix**
Closes #45 
Closes #44

### CODE REVIEW

File: src/jpipe_runner/framework/engine.py [hunk 1]
```diff
@@ -312,7 +333,32 @@ def get_execution_order(self) -> list[str]:
...
-            GLOBAL_LOGGER.error("Cycle detected in justification graph: %s", e)
+            # Try to find the cycle for a more precise error message
+            try:
+                cycle = next(nx.simple_cycles(self.graph))
+            except Exception:
+                cycle = None
+
+            if cycle:
+                cycle_labels = [self.graph.nodes[n].get("label", n) for n in cycle]
+                error_msg = (
```
[the remark this answers is not in the document]
Comment (BaptisteLacroix) [reply src/jpipe_runner/framework/engine.py:344]: issue created
